src/verse.zig

```diff
@@ -1,366 +1,49 @@
-                };
-                return;
-            };
-            stream.writevAll(vec) catch |err| switch (err) {
-                else => log.err("iovec write error Error {}", .{err}),
-            };
-            if (required > 2048) vrs.alloc.free(vecs);
-        },
-        else => unreachable,
-    }
-}
-
-/// sendRawSlice will allow you to send data directly to the client. It will not
-/// verify the current state, and will allow you to inject data into the HTTP
-/// headers. If you only want to send response body data, call quickStart() to
-/// send all headers to the client
-pub fn sendRawSlice(vrs: *Verse, slice: []const u8) NetworkError!void {
-    vrs.writeAll(slice) catch |err| switch (err) {
-        error.BrokenPipe => |e| return e,
-        else => unreachable,
-    };
-}
-
-/// Helper function to return a default error page for a given http status code.
-pub fn sendError(vrs: *Verse, comptime code: std.http.Status) !void {
-    return Router.defaultResponse(code)(vrs);
-}
-
-/// Takes a any object, that can be represented by json, converts it into a
-/// json string, and sends to the client.
-pub fn sendJSON(vrs: *Verse, json: anytype, comptime code: std.http.Status) !void {
-    if (code == .no_content) {
-        @compileError("Sending JSON is not supported with status code no content");
-    }
-
-    vrs.status = code;
-    vrs.content_type = .{
-        .base = .{ .application = .json },
-        .parameter = .@"utf-8",
-    };
-
-    try vrs.quickStart();
-    const data = std.json.stringifyAlloc(vrs.alloc, json, .{
-        .emit_null_optional_fields = false,
-    }) catch |err| {
-        log.err("Error trying to print json {}", .{err});
-        return error.Unknown;
-    };
-    vrs.writeAll(data) catch |err| switch (err) {
-        error.BrokenPipe => |e| return e,
-        else => unreachable,
-    };
-}
-
-/// This function may be removed in the future
-pub fn quickStart(vrs: *Verse) NetworkError!void {
-    if (vrs.status == null) vrs.status = .ok;
-    switch (vrs.downstream) {
-        .http, .zwsgi => |_| {
-            vrs.sendHeaders() catch |err| switch (err) {
-                error.BrokenPipe => |e| return e,
-                else => unreachable,
-            };
-
-            vrs.writeAll("\r\n") catch |err| switch (err) {
-                error.BrokenPipe => |e| return e,
-                else => unreachable,
-            };
-        },
-        else => unreachable,
-    }
-}
-
-test "Verse" {
-    std.testing.refAllDecls(@This());
-}
-
-const std = @import("std");
-const Allocator = std.mem.Allocator;
-const eql = std.mem.eql;
-const Stream = std.net.Stream;
-const AnyWriter = std.io.AnyWriter;
-const bufPrint = std.fmt.bufPrint;
-const allocPrint = std.fmt.allocPrint;
-const splitScalar = std.mem.splitScalar;
-const log = std.log.scoped(.Verse);
-const iovec = std.posix.iovec;
-const iovec_c = std.posix.iovec_const;
+const frame = @import("frame.zig");
```

Comment:
sorry I'm gonna be pendantic about this (I want to preserve as much as the git blame as I can)

Lets rename this file to root.zig

and then in a 2nd commit rename it back to verse.zig, that way git will create a new tree for it

Reply:
and then most of these should be TitleCase